x-pack/plugins/security_solution/public/detections/components/rules/query_bar/index.tsx

```diff
@@ -86,6 +88,13 @@ export const QueryBarDefineRule = ({
 
   const savedQueryServices = useSavedQueryServices();
 
+  useEffect((): void => {
+    if (onError != null) {
+      const errorExists = isInvalid || (errorMessage != null && errorMessage.trim() === '');
```

Comment:
No, I made the check purely based off the type to be sure there wasn't a case where it could have an array with an empty string and block preview, but you're right, don't think that's necessary. I updated to rely just on `isInvalid` for non-eql query bar and `isValid` for eql query bar.